### PR TITLE
make 'compile on save without preview' work with 'autosave'

### PR DESCRIPTION
**Pull request by tmandke**

### CODE REVIEW

**Comment by tmandke** [comment]
This PR preserves current behaviour of 'compile on save'.
Correct me if I am wrong. The intention of 'compile on save' is to save the file when you run the atom command 'coffee-compile:compile'. If so, then would it not be more clear for it to be 'save on compile' ??

File: spec/fs-util-spec.coffee [hunk 1]
```diff
@@ -1,3 +1,4 @@
+fs = require 'fs'
```
Comment (adrianlee44) [inline spec/fs-util-spec.coffee]: `fs` is not used here

File: lib/main.coffee [hunk 2]
```diff
@@ -36,17 +36,20 @@ module.exports =
       description: 'Remove all path parts'
 
   activate: ->
-    saveDisposable = null
+    saveDisposables = null
```
Comment (adrianlee44) [inline lib/main.coffee]: Since `saveDisposables` should be an array, just initialize as empty array here

File: lib/main.coffee [hunk 3]
```diff
@@ -36,17 +36,20 @@ module.exports =
...
 
     atom.commands.add 'atom-workspace', 'coffee-compile:compile': => @display()
 
     atom.config.observe 'coffee-compile.compileOnSaveWithoutPreview', (value) =>
       if not value and saveDisposable?
```
Comment (adrianlee44) [inline lib/main.coffee]: no need for `saveDisposable?` check here

File: lib/main.coffee [hunk 4]
```diff
@@ -36,17 +36,20 @@ module.exports =
...
-        saveDisposable.dispose()
-        saveDisposable = null
+        sd.dispose() for sd in saveDisposables
+        saveDisposables = null
 
       else if value
-        saveDisposable = atom.commands.add 'atom-workspace', 'core:save': => @save()
+        saveDisposables = []
+        saveDisposables << atom.workspace.observeTextEditors (editor) =>
```
Comment (adrianlee44) [inline lib/main.coffee]: Why are you doing a bitwise left shift here?

File: spec/fs-util-spec.coffee (hunk 1 above)
Comment (adrianlee44) [inline spec/fs-util-spec.coffee]: nvm. Not sure how the tests were passing before.

File: lib/main.coffee (hunk 4 above)
Comment (tmandke) [inline lib/main.coffee]: oops too use to ruby

**Comment by adrianlee44** [comment]
@tmandke, thanks for the PR. merging.

**Pull request merged**